Jarvis fail on help command

Hey,
So I was trying this tool and I notice `help` command will break the setup... I have ubuntu 18.04 and python3.6.9, and I'm running `./jarvis`.

```
Jarvis' sound is by default disabled.
In order to let Jarvis talk out loud type: enable sound
Type 'help' for a list of available actions.
163 plugins loaded 6 plugins disabled. More information: status

~> Hi, what can I do for you?

I could not identify your command...
~> What can I do for you?
elp
I could not identify your command...
~> What can I do for you?
fasdfdas
I could not identify your command...
~> What can I do for you?
help

Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/bersamk/workspace/Jarvis/jarviscli/__main__.py", line 22, in <module>
    main()
  File "/home/bersamk/workspace/Jarvis/jarviscli/__main__.py", line 17, in main
    jarvis.executor(command)
  File "/home/bersamk/workspace/Jarvis/jarviscli/Jarvis.py", line 189, in executor
    self.cmdloop()
  File "/usr/lib/python3.6/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/usr/lib/python3.6/cmd.py", line 217, in onecmd
    return func(arg)
  File "/home/bersamk/workspace/Jarvis/jarviscli/CmdInterpreter.py", line 463, in do_help
    print_say("", self)
  File "/home/bersamk/workspace/Jarvis/jarviscli/utilities/GeneralUtilities.py", line 28, in print_say
    self.speech.text_to_speech(text)
  File "/home/bersamk/workspace/Jarvis/jarviscli/utilities/voice.py", line 66, in text_to_speech
    tts = gTTS(speech, lang="en")
  File "/home/bersamk/workspace/Jarvis/env/lib/python3.6/site-packages/gtts/tts.py", line 120, in __init__
    assert text, 'No text to speak'
AssertionError: No text to speak
```

--- a/jarviscli/CmdInterpreter.py
+++ b/jarviscli/CmdInterpreter.py
@@ -423,7 +423,7 @@ class CmdInterpreter(Cmd):
         if self._api.is_spinner_running():
             self._api.spinner_stop('Some error has occured')
 
-        self.say("Goodbye, see you later!", self, Fore.RED)
+        self.say("Goodbye, see you later!", Fore.RED)
         self.scheduler.stop_all()
         sys.exit()
 
@@ -449,7 +449,7 @@ class CmdInterpreter(Cmd):
                 count_disabled))
 
         if "short" not in s and count_disabled > 0:
-            self.say("", self)
+            self.say("")
             for disabled, reason in self._plugin_manager.get_disabled().items():
                 self.say(
                     "{:<20}: {}".format(

--- a/jarviscli/CmdInterpreter.py
+++ b/jarviscli/CmdInterpreter.py
@@ -11,7 +11,7 @@ from PluginManager import PluginManager
 from utilities import schedule
 from utilities.voice import create_voice
 from utilities.notification import notify
-from utilities.GeneralUtilities import print_say, get_parent_directory
+from utilities.GeneralUtilities import get_parent_directory
 
 from packages.memory.memory import Memory
 from utilities.animations import SpinnerThread
@@ -327,6 +327,7 @@ class CmdInterpreter(Cmd):
         self.memory = Memory()
         self.scheduler = schedule.Scheduler()
         self._api = JarvisAPI(self)
+        self.say = self._api.say
 
         # Remember voice settings
         self.enable_voice = self._api.get_data('enable_voice')
@@ -336,13 +337,14 @@ class CmdInterpreter(Cmd):
             self.speech_rate = 120
 
         # what if the platform does not have any engines, travis doesn't have sapi5 acc to me
+
         try:
             gtts_status = self._api.get_data('gtts_status')
             self.speech = create_voice(
                 self, gtts_status, rate=self.speech_rate)
         except Exception as e:
-            print_say("Voice not supported", self, Fore.RED)
-            print_say(str(e), self, Fore.RED)
+            self.say("Voice not supported", Fore.RED)
+            self.say(str(e), Fore.RED)
 
         self.fixed_responses = {"what time is it": "clock",
                                 "where am i": "pinpoint",
@@ -421,7 +423,7 @@ class CmdInterpreter(Cmd):
         if self._api.is_spinner_running():
             self._api.spinner_stop('Some error has occured')
 
-        print_say("Goodbye, see you later!", self, Fore.RED)
+        self.say("Goodbye, see you later!", self, Fore.RED)
         self.scheduler.stop_all()
         sys.exit()
 
@@ -431,7 +433,7 @@ class CmdInterpreter(Cmd):
 
     def error(self):
         """Jarvis let you know if an error has occurred."""
-        print_say("I could not identify your command...", self, Fore.RED)
+        self.say("I could not identify your command...", Fore.RED)
 
     def interrupt_handler(self, signal, frame):
         """Closes Jarvis on SIGINT signal. (Ctrl-C)"""
@@ -441,30 +443,28 @@ class CmdInterpreter(Cmd):
         """Prints plugin status status"""
         count_enabled = self._plugin_manager.get_number_plugins_loaded()
         count_disabled = len(self._plugin_manager.get_disabled())
-        print_say(
+        self.say(
             "{} Plugins enabled, {} Plugins disabled.".format(
                 count_enabled,
-                count_disabled),
-            self)
+                count_disabled))
 
         if "short" not in s and count_disabled > 0:
-            print_say("", self)
+            self.say("", self)
             for disabled, reason in self._plugin_manager.get_disabled().items():
-                print_say(
+                self.say(
                     "{:<20}: {}".format(
                         disabled,
-                        " OR ".join(reason)),
-                    self)
+                        " OR ".join(reason)))
 
     def do_help(self, arg):
         if arg:
             Cmd.do_help(self, arg)
         else:
-            print_say("", self)
+            self.say("")
             headerString = "These are valid commands for Jarvis"
             formatString = "Format: command ([aliases for command])"
-            print_say(headerString, self)
-            print_say(formatString, self, Fore.BLUE)
+            self.say(headerString)
+            self.say(formatString, Fore.BLUE)
             pluginDict = self._plugin_manager.get_plugins()
             uniquePlugins = {}
             for key in pluginDict.keys():
@@ -488,5 +488,5 @@ class CmdInterpreter(Cmd):
             Cmd.columnize(self, helpOutput)
 
     def help_status(self):
-        print_say("Prints info about enabled or disabled plugins", self)
-        print_say("Use \"status short\" to omit detailed information.", self)
+        self.say("Prints info about enabled or disabled plugins")
+        self.say("Use \"status short\" to omit detailed information.")

--- a/jarviscli/packages/forecast.py
+++ b/jarviscli/packages/forecast.py
@@ -1,10 +1,11 @@
 # -*- coding: utf-8 -*-
-from utilities.GeneralUtilities import print_say
-from colorama import Fore
 import json
-from . import mapps
 import requests
+
+from colorama import Fore
+
 from utilities.dateTime import WeekDay
+from . import mapps
 
 
 def main(jarvis, s):

--- a/jarviscli/utilities/GeneralUtilities.py
+++ b/jarviscli/utilities/GeneralUtilities.py
@@ -1,10 +1,11 @@
 # -*- coding: utf-8 -*-
-import sys
-import os
-from colorama import Fore
 import distutils.spawn
+import os
 from platform import win32_ver
+import sys
+import warnings
 
+from colorama import Fore
 
 MACOS = 'darwin'
 WIN = 'win32'
@@ -22,7 +23,15 @@ def print_say(text, self, color=""):
         :param text: the text to print (or talk)
                color: Fore.COLOR (ex Fore.BLUE), color for text
         :return: Nothing to return.
-        """
+
+        .. deprecated::
+            Use ``JarvisAPI.say(text, color="", speak=True))`` instead.
+    """
+    warnings.warn(
+        "GeneralUtilities.print_say is deprecated now and will be \
+        removed in the future. Please use \
+        ``JarvisAPI.say(text, color=\"\", speak=True))`` instead.",
+        DeprecationWarning)
     print(color + text + Fore.RESET)
     if self.enable_voice:
         self.speech.text_to_speech(text)
@@ -55,11 +64,14 @@ def unsupported(platform, silent=False):
         def wrapped(*args, **kwargs):
             if sys.platform == platform:
                 if not silent:
-                    print('{}Command is unsupported for platform `{}`{}'.
-                          format(Fore.RED, sys.platform, Fore.RESET))
+                    print(
+                        '{}Command is unsupported for platform `{}`{}'.format(
+                            Fore.RED, sys.platform, Fore.RESET))
             else:
                 func(*args, **kwargs)
+
         return wrapped
+
     return noop_wrapper
 
 

